### PR TITLE
refactor build image with common action.yml

### PR DESCRIPTION
**Pull request by chensuyue**
## Description

refactor build image with common action.yml

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [x] Others (enhancement, documentation, validation, etc.)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: .github/actions/image-build/action.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      shell: bash
+      run: |
+        cp ${{ inputs.docker_compose_path }} ${{ inputs.work_dir }}/docker-compose.yaml
+        REGISTRY=${{ inputs.registry }} \
+        docker compose -p ${GITHUB_RUN_NUMBER} build --no-cache
+      working-directory: ${{ inputs.work_dir }}
+    - name: Push Containers
+      shell: bash
+      run: |
+        REGISTRY=${{ inputs.registry }} \
```
Comment (daisy-ycguo) [inline .github/actions/image-build/action.yml:43]: I wonder if we want to allow to push to multiple registries.

File: .github/actions/image-build/action.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2024 Intel Corporation
```
Comment (daisy-ycguo) [inline .github/actions/image-build/action.yml:1]: Is it possible to put actions/image-build/action.yml to Validate repo? Then all repo can leverage this action.

**Comment by chensuyue** [comment]
Test pass: https://github.com/opea-project/GenAIComps/actions/runs/10193674110/job/28198729619?pr=385

**Review by ZePan110 — APPROVED** [review]

**Comment by chensuyue** [comment]
refer action from: https://github.com/opea-project/Validation/blob/main/actions/image-build/action.yml

**Review by daisy-ycguo — APPROVED** [review]
lgtm

**Pull request merged**